### PR TITLE
fix: adapt to new behavior of usethis::use_news_md()

### LINKED ISSUE #679: failures with dev usethis & testthat
I am really surprised at the deletions... trying with dev testthat soon.

```r
==> devtools::test()

ℹ Testing fledge
Starting 2 test processes
✔ | F W S  OK | Context
✔ |         1 | commits [0.5s]                           
✖ | 1       0 | demo [21.8s]                             
─────────────────────────────────────────────────────────
Failure (test-demo.R:2:3): demo vignette works
Snapshot of `path` to 'demo/demo.md' has changed
Run `testthat::snapshot_review('demo/')` to review changes
Backtrace:
    ▆
 1. └─fledge:::test_galley("demo") at test-demo.R:2:2
 2.   └─testthat::expect_snapshot_file(path, name = md_name, compare = testthat::compare_file_text) at tests/testthat/helper-vignettes.R:108:2
─────────────────────────────────────────────────────────
✖ | 2       2 | finalize-version [34.9s]                 
─────────────────────────────────────────────────────────
Failure (test-finalize-version.R:18:3): finalize_version(push = FALSE)
Snapshot of "NEWS.md" to 'finalize-version/NEWS-push-false.md' has changed
Run `testthat::snapshot_review('finalize-version/')` to review changes

Failure (test-finalize-version.R:43:3): finalize_version(push = TRUE)
Snapshot of "NEWS.md" to 'finalize-version/NEWS-push-true.md' has changed
Run `testthat::snapshot_review('finalize-version/')` to review changes
─────────────────────────────────────────────────────────
✔ |        10 | fledgling [0.7s]                         
✖ | 1      10 | parse-news-items [17.8s]                 
─────────────────────────────────────────────────────────
Failure (test-parse-news-items.R:32:3): Will use commits
Snapshot of "NEWS-merge.md" to 'parse-news-items/NEWS-merge.md' has changed
Run `testthat::snapshot_review('parse-news-items/')` to review changes
─────────────────────────────────────────────────────────
✖ | 3      19 | bump_version [84.5s]                     
─────────────────────────────────────────────────────────
Failure (test-bump_version.R:15:3): bump_version() works -- dev
Snapshot of "NEWS.md" to 'bump_version/NEWS.md' has changed
Run `testthat::snapshot_review('bump_version/')` to review changes

Failure (test-bump_version.R:25:3): bump_version() works -- dev
Snapshot of "NEWS.md" to 'bump_version/NEWS2.md' has changed
Run `testthat::snapshot_review('bump_version/')` to review changes

Failure (test-bump_version.R:41:3): bump_version() works -- not dev
Snapshot of "NEWS.md" to 'bump_version/NEWS-nondev.md' has changed
Run `testthat::snapshot_review('bump_version/')` to review changes
─────────────────────────────────────────────────────────
✔ |         4 | tag_version [17.4s]                      
✖ | 1       1 | unbump_version [20.4s]                   
─────────────────────────────────────────────────────────
Failure (test-unbump_version.R:32:3): unbump_version() works
Snapshot of `file.path(repo, "NEWS.md")` to 'unbump_version/NEWS.md' has changed
Run `testthat::snapshot_review('unbump_version/')` to review changes
─────────────────────────────────────────────────────────
✔ |         1 | with_demo_project                        
✔ |        11 | update-news [20.6s]                      
                                                         
══ Results ══════════════════════════════════════════════
Duration: 115.6 s

[ FAIL 8 | WARN 0 | SKIP 0 | PASS 59 ]
Deleting unused snapshots:
• bump_version/NEWS-nondev.md
• bump_version/NEWS-nondev.new.md
• bump_version/NEWS.md
• bump_version/NEWS.new.md
• bump_version/NEWS2.md
• bump_version/NEWS2.new.md
• finalize-version/NEWS-push-false.md
• finalize-version/NEWS-push-false.new.md
• finalize-version/NEWS-push-true.md
• finalize-version/NEWS-push-true.new.md
• parse-news-items/NEWS-merge.md
• parse-news-items/NEWS-merge.new.md
• parse-news-items/NEWS.md
• unbump_version/NEWS.md
• unbump_version/NEWS.new.md
• update-news/newchangelog.md
• update-news/samedev-base.md
• update-news/samedev-updated.md
• update-news/samedev.md
```

### PR DESCRIPTION
**Pull request by maelle**
Fix #679

### CODE REVIEW

File: tests/testthat/test-bump_version.R [hunk 1]
```diff
@@ -1,4 +1,5 @@
 test_that("bump_version() works -- dev", {
+  testthat::skip_if_offline() # because of usethis::use_news_md() -> available.packages()
```
Comment (maelle) [inline tests/testthat/test-bump_version.R:2]: as a reminder, `skip_if_offline()` also skips tests on CRAN.

File: R/fledgling.R [hunk 2]
```diff
@@ -198,7 +198,7 @@ write_fledgling <- function(fledgeling) {
 
   news_df <- fledgeling$news
   news_lines <- purrr::map_chr(
-    split(news_df, sort(as.numeric(rownames(news_df)))),
+    split(news_df, seq_len(nrow(news_df))),
```
[the remark this answers is not in the document]
Comment (krlmlr) [reply R/fledgling.R:201]: `vec_split()` might be better.

File: tests/testthat/_snaps/demo/demo.md [hunk 3]
```diff
@@ -136,7 +136,7 @@ Let's take a look at the contents:
 ```r
 news <- readLines(usethis::proj_path("NEWS.md"))
 cat(news, sep = "\n")
-## # tea 0.0.0.9000
+## # tea (development version)
```
Comment (krlmlr) [inline tests/testthat/_snaps/demo/demo.md:139]: Intended?

🙀 🙀 🙀

Comment (maelle) [reply tests/testthat/_snaps/demo/demo.md:139]: yes! but after the first bump it becomes an actual version. if you don't want this behavior, we should not create the NEWS file with usethis.

I find the new behavior more logical https://github.com/r-lib/usethis/issues/1708 :sweat_smile: :innocent:

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**